### PR TITLE
broker: increase nofile limit to avoid assertion failure in zuuid_new()

### PR DESCRIPTION
**Pull request by grondo**
As discussed in #2593, file descriptor exhaustion may cause the broker to hit an assertion failure in `zuuid_new()` due to failure to open `/dev/urandom`, likely due to many API users. (The reproducer creates 1000 simultaneous `flux mini run` invocations). 

Though not a real fix, increase the number of open file descriptor limit in the broker to the current hard limit. This avoids the need for users to remember to do this by hand if their use case hits this issue.

### CODE REVIEW

**Comment by garlick** [comment]
Oh good, I was thinking we should do this.

Hmm, I wonder if we could avoid this particular failure (not to mention the assertion in zuuid which is an unpleasant way to handle the error) by generating uuid's with sodium's RNG, e.g:

```c
uint8_t uuid[16];
randombytes_buf(buf, sizof (buf));
```

I think it opens /dev/urandom once and holds the fd in a global context (and is probably already doing so in the broker because zmq uses it to generate curve session keys).  Alternatively, it mentions the `getrandom()` syscall which it says is available in newer kernels.

https://libsodium.gitbook.io/doc/generating_random_data

Just thinking out - this PR Is fine as is.

**Comment by grondo** [comment]
That's a good idea. Would it be overkill to make a drop-in replacement for zuuid and plop it in libutil?

**Comment by garlick** [comment]
There's an RFC for uuids

https://tools.ietf.org/html/rfc4122

We should make sure we're aware of that before we provide a new implementation, or at least do nothing dumber than what zuuid is doing.

On the fence on whether the zuuid API is a good thing.  The malloc() implicit in `zuuid_new()` (and required error handling code when using it) could be avoided if we changed the api a bit.  Maybe some kind of libutil helper though?  There's also <uuid/uuid.h> from `uuid-dev`.  We could require that and just provide functions that deal with  static `uuid_t`'s?  Or just use `uuid_generate()` (but I think it opens /dev/urandom at least on the fist call?)

File: src/broker/broker.c [hunk 1]
```diff
@@ -281,6 +282,26 @@ static int setup_profiling (const char *program, int rank)
     return (0);
 }
 
+static int increase_rlimits (void)
+{
+    struct rlimit rlim;
+
+    /*  Increase number of open files to max to prevent potential failures
+     *   due to file descriptor exhaustion (e.g. zuuid_new() failure to
+     *   open /dev/urandom
```
Comment (chu11) [inline src/broker/broker.c]: nit: end parentheses :P

Comment (grondo) [reply src/broker/broker.c]: Thanks! Good catch. I pushed a fixup just in time. :-)

**Pull request merged**